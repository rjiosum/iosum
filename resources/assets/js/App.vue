<template>
    <v-app>
        <AppNotify />
        <AppSideBar v-if="isAuthenticated" />
        <AppToolBar v-if="isAuthenticated" />
        <v-main>
            <router-view :key="$route.fullPath"></router-view>
        </v-main>
        <AppToTop />
    </v-app>
</template>

<script>
    import AppNotify from "./components/core/AppNotify";
    import AppSideBar from "./components/core/AppSideBar";
    import AppToolBar from "./components/core/AppToolBar";
    import AppToTop from "./components/core/AppToTop";
    import {mapGetters} from 'vuex'

    export default {
        name: "App",
        components: {
            AppNotify,
            AppSideBar,
            AppToolBar,
            AppToTop
        },
        computed:{
            ...mapGetters({
                isAuthenticated: 'auth/isAuthenticated'
            })
        }
    }
</script>