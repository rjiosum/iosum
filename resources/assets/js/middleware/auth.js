export default function auth ({ next, store }){
    if(!store.getters['auth/isAuthenticated']){
        return next({
            name: 'login'
        });
    }
    return next();
}