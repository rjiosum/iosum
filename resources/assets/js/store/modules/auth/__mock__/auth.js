import Vue from 'vue'
import Vuex from 'vuex'
import * as types from "../mutation-types"

Vue.use(Vuex)

export const state = {
    user: {
        "first_name": "John",
        "last_name": "Doe",
        "name": "John Doe",
        "email": "john@gmail.com",
        "avatar": "/storage/admin/avatars/avatar.png"
    }
}

export const getters = {
    user: jest.fn().mockReturnValue({
        "first_name": "John",
        "last_name": "Doe",
        "name": "John Doe",
        "email": "john@gmail.com",
        "avatar": "/storage/admin/avatars/avatar.png"
    }),
    isAuthenticated: jest.fn().mockReturnValue(true)
}

export const mutations = {
    [types.LOGIN]: jest.fn(),
    [types.LOGOUT]: jest.fn()
}

export const actions = {
    login: jest.fn(),
    logout: jest.fn(),
    getUser: jest.fn()
}



export function __createMocks(custom = { getters: {}, mutations: {}, actions: {}, state: {} }) {
    const mockGetters = Object.assign({}, getters, custom.getters);
    const mockMutations = Object.assign({}, mutations, custom.mutations);
    const mockActions = Object.assign({}, actions, custom.actions);
    const mockState = Object.assign({}, state, custom.state);

    return {
        getters: mockGetters,
        mutations: mockMutations,
        actions: mockActions,
        state: mockState,
        store: new Vuex.Store({
            modules: {
                auth: {
                    getters: mockGetters,
                    mutations: mockMutations,
                    actions: mockActions,
                    state: mockState,
                    namespaced: true
                }
            }
        }),
    };
}

//export const store = __createMocks().store;