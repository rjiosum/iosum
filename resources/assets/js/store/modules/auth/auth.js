import {Factory} from '@/repositories/factory'
import * as types from './mutation-types'

const AuthRepository = Factory.get('auth')
const UserRepository = Factory.get('user')

const state = () => ({
    user: null
})

const getters = {
    user(state){
         return state.user
    },
    isAuthenticated(state){
        return !!state.user
    }
}

const mutations = {
    [types.LOGIN](state, user){
        state.user = user
    },
    [types.LOGOUT](state){
        state.user = null
    }
}

const actions = {
    async login({commit, dispatch}, credentials){
        const response = await AuthRepository.login(credentials)
        if(response.data.status){
            commit(types.LOGIN, response.data.data);
            dispatch('navbar/getNavBar', {}, {root:true});
        }
        return response
    },
    async logout({commit}){
        const response = await AuthRepository.logout()
        if(response.data.status){
            commit(types.LOGOUT);
        }
        return response
    },
    async getUser ({commit}) {
        const response = await UserRepository.getUser()
        if(response.data.status){
            commit(types.LOGIN, response.data.data);
        }else{
            commit(types.LOGOUT);
        }
        return response
    }
}


export default {
    namespaced: true,
    state,
    getters,
    mutations,
    actions
}