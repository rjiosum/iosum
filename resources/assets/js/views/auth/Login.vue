<template>
    <v-container
            class="fill-height"
            fluid
    >
        <v-row no-gutters align="center" justify="center">
            <v-col
                    cols="12"
                    sm="8"
                    md="4"
            >
                <v-form @submit.prevent="login">
                    <AppCard class="px-5 py-3 mx-auto" max-width="374">
                        <template v-slot:heading>
                            <div class="display-1 font-weight-bold text-center">
                                Login
                            </div>
                            <div class="subtitle-1 font-weight-light text-center">
                                Please enter your credentials
                            </div>
                        </template>

                        <v-card-text>
                            <v-text-field
                                    color="purple"
                                    label="E-mail"
                                    placeholder="Enter your e-mail"
                                    v-model.trim="form.email"
                                    type="email"
                                    name="email"
                                    prepend-icon="mdi-email-outline"
                                    :error-messages="emailErrors"
                                    @input="$v.form.email.$touch()"
                                    @blur="$v.form.email.$touch()"
                                    class="pt-lg-10 pt-md-10 pt-sm-5"
                            ></v-text-field>
                            <v-text-field
                                    color="purple"
                                    label="Password"
                                    placeholder="Enter your password"
                                    v-model.trim="form.password"
                                    :type="showPassword ? 'text' : 'password'"
                                    name="password"
                                    prepend-icon="mdi-lock-open"
                                    :append-icon="showPassword ? 'mdi-eye' : 'mdi-eye-off'"
                                    @click:append="showPassword = !showPassword"
                                    :error-messages="passwordErrors"
                                    @input="$v.form.password.$touch()"
                                    @blur="$v.form.password.$touch()"
                                    class="pt-lg-10 pt-md-10 pt-sm-5"
                            ></v-text-field>

                            <v-switch color="purple" class="pt-5" v-model="form.remember"
                                      :label="`Stay signed in`"></v-switch>

                        </v-card-text>

                        <template v-slot:actions>
                            <v-spacer></v-spacer>
                            <v-btn depressed
                                   color="accent"
                                   type="submit"
                                   width="50%"
                                   :loading="submitted"
                                   :disabled="submitted">Login
                            </v-btn>
                        </template>
                    </AppCard>
                </v-form>
            </v-col>
        </v-row>
    </v-container>


</template>

<script>
    import {validationMixin} from 'vuelidate'
    import {required, email} from 'vuelidate/lib/validators'
    import {mapGetters} from "vuex";
    import AppCard from "../../components/core/AppCard";

    export default {
        name: "AppLogin",
        mixins: [validationMixin],
        validations: {
            form: {
                email: {required, email},
                password: {required}
            }
        },
        components: {
            AppCard
        },
        data() {
            return {
                showPassword: false,
                form: {
                    email: null,
                    password: null,
                    remember: false
                },
                submitted: false
            }
        },
        methods: {
            async login() {
                this.$v.$touch();
                if (this.$v.$invalid) {
                    return;
                }
                this.submitted = true;
                try {
                    const response = await this.$store.dispatch('auth/login', this.form)

                    if (response.data.status) {
                        await this.$store.dispatch('notify/setNotice', {
                            msg: "Successfully logged in.",
                            color: "success",
                            icon: "mdi-check-circle"
                        });
                        if (this.$route.query.redirect) {
                            await this.$router.replace(this.$route.query.redirect);
                            return;
                        }
                        await this.$router.replace({name: 'dashboard'});
                    }
                } catch (e) {
                    //console.log(e)
                    this.submitted = false;
                }

            }
        },
        computed: {
            ...mapGetters({errors: "errors/errors"}),
            emailErrors() {
                const errors = [];
                if (!this.$v.form.email.$dirty) return errors;
                !this.$v.form.email.email && errors.push('Must be valid e-mail');
                !this.$v.form.email.required && errors.push('E-mail is required');
                this.errors.email && errors.push(this.errors.email[0]);
                return errors;
            },
            passwordErrors() {
                const errors = [];
                if (!this.$v.form.password.$dirty) return errors;
                !this.$v.form.password.required && errors.push('Password is required');
                this.errors.password && errors.push(this.errors.password[0]);
                return errors;
            }
        }
    }
</script>