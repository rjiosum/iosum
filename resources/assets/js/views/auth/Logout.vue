<template>
    <div></div>
</template>

<script>
    export default {
        name: "Logout",
        created() {
            EventBus.$emit('kick')
        }
    }
</script>


