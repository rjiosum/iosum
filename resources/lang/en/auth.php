<?php

return [
    'login' => 'Successfully logged in.',
    'logout' => 'Successfully logged out.',
];
