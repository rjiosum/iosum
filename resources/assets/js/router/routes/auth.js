import {lazy, prefix} from '../utils/utility';
import guest from '@/middleware/guest';
import auth from '@/middleware/auth';
import isLoggedIn from '@/middleware/isLoggedIn';

export default [
    {
        path: prefix('login'),
        name: 'login',
        component: lazy('auth/Login'),
        meta: {
            middleware: [isLoggedIn, guest],
            title: 'Login'
        }
    },
    {
        path: prefix('logout'),
        name: 'logout',
        component: lazy('auth/Logout'),
        meta: {
            middleware: [isLoggedIn, auth]
        }
    }
]
