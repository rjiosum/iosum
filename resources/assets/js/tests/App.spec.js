import App from '@/App.vue'
import VueRouter from 'vue-router'
import Vuex from 'vuex'
import {shallowMount, createLocalVue} from '@vue/test-utils'
import Vuetify from "vuetify";

import { __createMocks as createStoreMocks } from '@/store/modules/auth/__mock__/auth';

describe('App.vue', () => {
    let storeMocks = null;
    let localVue = null;
    let wrapper = null;
    let vuetify = null;
    let router = null;

    beforeEach(() => {

        localVue = createLocalVue();
        localVue.use(VueRouter);
        localVue.use(Vuex);
        storeMocks = createStoreMocks();

        router = new VueRouter()
        vuetify = new Vuetify()

        wrapper = shallowMount(App, {
            store: storeMocks.store,
            localVue,
            router,
            vuetify
        })
    })

    test('snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('can find AppNotify component', () => {
        const component = wrapper.findComponent({name: 'AppNotify'})
        expect(component.exists()).toBe(true)
    })

    test('can find AppSideBar component', () => {
        const component = wrapper.findComponent({name: 'AppSideBar'})
        expect(component.exists()).toBe(true)
    })

    test('can find AppToolBar component', () => {
        const component = wrapper.findComponent({name: 'AppToolBar'})
        expect(component.exists()).toBe(true)
    })

    test('can find AppToTop component', () => {
        const component = wrapper.findComponent({name: 'AppToTop'})
        expect(component.exists()).toBe(true)
    })

})


