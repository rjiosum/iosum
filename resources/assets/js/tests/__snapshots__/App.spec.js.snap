// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue snapshot 1`] = `
VueWrapper {
  "_emitted": Object {
    "hook:beforeCreate": Array [
      Array [],
    ],
    "hook:beforeMount": Array [
      Array [],
    ],
    "hook:created": Array [
      Array [],
    ],
    "hook:mounted": Array [
      Array [],
    ],
  },
  "_emittedByOrder": Array [
    Object {
      "args": Array [],
      "name": "hook:beforeCreate",
    },
    Object {
      "args": Array [],
      "name": "hook:created",
    },
    Object {
      "args": Array [],
      "name": "hook:beforeMount",
    },
    Object {
      "args": Array [],
      "name": "hook:mounted",
    },
  ],
  "isFunctionalComponent": undefined,
}
`;
